src/Test.UnitTests.Sarif.WorkItems/SarifWorkItemExtensionsTests.cs

```diff
@@ -52,26 +53,109 @@ public void SarifWorkItemExtensions_CreateWorkItemTitle_HandlesSingleResultWithR
             sb.Length.Should().Be(0, because: Environment.NewLine + sb.ToString());
         }
 
+        private class PhraseToolNamesTestCase
+        {
+            public PhraseToolNamesTestCase(List<string> input, string expectedOutput)
+            {
+                Input = input;
+                ExpectedOutput = expectedOutput;
+            }
+
+            public List<string> Input { get; }
+            public string ExpectedOutput { get; }
+        }
+
+        private static readonly ReadOnlyCollection<PhraseToolNamesTestCase> s_phraseToolNamesTestCases =
+            new ReadOnlyCollection<PhraseToolNamesTestCase>(new PhraseToolNamesTestCase[]
+            {
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {""},
+                    expectedOutput: ""),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"CredentialScanner"},
+                    expectedOutput: "'CredentialScanner'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"CredentialScanner", "Binskim"},
+                    expectedOutput: "'CredentialScanner' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"SEMMLE", "CredentialScanner", "Binskim"},
+                    expectedOutput: "'SEMMLE', 'CredentialScanner' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"SEMMLE", "", "Binskim"},
+                    expectedOutput: "'SEMMLE' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"", "CredentialScanner", "SEMMLE", "Binskim"},
+                    expectedOutput: "'CredentialScanner', 'SEMMLE' and 'Binskim'"),
+
+                new PhraseToolNamesTestCase(
+                    input: new List<string>() {"", "CredentialScanner"},
+                    expectedOutput: "'CredentialScanner'"),
+            });
+
         [Fact]
-        public void SarifWorkItemExtensions_ComputeToolResultCounts_CountsSingleResult()
+        public void SarifWorkItemExtensions_PhraseToolNames_ConstructPhraseCorrectly()
         {
+            StringBuilder sb = new StringBuilder();
+
+            foreach (PhraseToolNamesTestCase testCase in s_phraseToolNamesTestCases)
+            {
+                string actualOutput = SarifWorkItemsExtensions.PhraseToolNames(testCase.Input);
+
+                bool succeeded = (testCase.ExpectedOutput == null && actualOutput == null)
+                    || (actualOutput?.Equals(testCase.ExpectedOutput, StringComparison.Ordinal) == true);
+
+                if (!succeeded)
+                {
+                    sb.AppendLine($"    Input: {testCase.Input} Expected: {testCase.ExpectedOutput} Actual: {actualOutput}");
+                }
+            }
+            sb.Length.Should().Be(0,
```

Comment:
>s [](start = 12, length = 1)

I know you copied the example I showed you. Be aware that the style guidelines will want you to leave a blank line after a closing brace.